views/js/payment-components/card-payment-method.js

```diff
@@ -253,4 +268,22 @@ jQuery(function ($) {
         data = '';
         allValidCard = false;
     }
+
+    function isPlaceOrderInProgress() {
+        return placeOrderInProgress;
+    }
+
+    function placingOrderStarts()
+    {
+        placeOrderInProgress = true;
+        paymentForm.find('button[type="submit"]').prop('disabled', true);
+        paymentForm.find('button[type="submit"] i').toggleClass('icon-spinner icon-chevron-right right');
+    }
+
+    function placingOrderEnds()
+    {
+        placeOrderInProgress = false;
+        paymentForm.find('button[type="submit"]').prop('disabled', false);
+        paymentForm.find('button[type="submit"] i').toggleClass('icon-spinner icon-chevron-right right');
+    }
```

Comment:
No way to have these functions in a common place?

Reply:
Couldn't figure it out how to require a PrestaShop registered JS loaded for sure before the function from that one is called that's why the duplications.